x-pack/plugins/enterprise_search/public/applications/analytics/components/add_analytics_collections/add_analytics_collection_logic.ts

```diff
@@ -0,0 +1,115 @@
+
+export interface AddAnalyticsCollectionsActions {
+  apiError(error: HttpError): HttpError;
+  apiSuccess(collection: AnalyticsCollection): AnalyticsCollection;
+  createAnalyticsCollection(): void;
+  makeRequest: typeof AddAnalyticsCollectionsAPILogic.actions.makeRequest;
+  setInputError(error: string | boolean): { inputError: string | boolean };
+  setNameValue(name: string): { name: string };
+}
+
+interface AddAnalyticsCollectionValues {
+  canSubmit: boolean;
+  hasInputError: boolean;
+  inputError: string | boolean;
+  isLoading: boolean;
+  name: string;
+  status: typeof AddAnalyticsCollectionsAPILogic.values.status;
+}
+
+export const AddAnalyticsCollectionLogic = kea<
+  MakeLogicType<AddAnalyticsCollectionValues, AddAnalyticsCollectionsActions>
+>({
+  actions: {
+    createAnalyticsCollection: () => {},
+    setInputError: (inputError: string | boolean) => ({ inputError }),
+    setNameValue: (name: string) => ({ name }),
+  },
+  connect: {
+    actions: [AddAnalyticsCollectionsAPILogic, ['apiError', 'apiSuccess', 'makeRequest']],
+    values: [AddAnalyticsCollectionsAPILogic, ['status']],
+  },
+  listeners: ({ values, actions }) => ({
+    apiError: (error) => flashAPIErrors(error),
+    apiSuccess: async ({ name }, breakpoint) => {
+      // Wait for propagation of the new collection
+      flashSuccessToast(
+        i18n.translate('xpack.enterpriseSearch.analytics.collectionsCreate.action.successMessage', {
+          defaultMessage: "Successfully added collection '{name}'",
+          values: {
+            name,
+          },
+        })
+      );
+      await breakpoint(1000);
+      KibanaLogic.values.navigateToUrl(
+        generateEncodedPath(COLLECTION_VIEW_PATH, {
+          name,
+        })
+      );
+    },
+    createAnalyticsCollection: () => {
+      const { name } = values;
+      actions.makeRequest({ name });
+    },
+    setNameValue: ({ name }) => {
+      if (!isAlphaNumericOrUnderscore(name)) {
+        const message = i18n.translate(
+          'xpack.enterpriseSearch.analytics.collectionsCreate.action.invalidCollectionName',
+          {
+            defaultMessage: 'Name must only contain alphanumeric characters and underscores',
+          }
+        );
+        return actions.setInputError(message);
+      }
+      return actions.setInputError(false);
+    },
+  }),
+  path: ['enterprise_search', 'add_analytics_collection'],
+  reducers: {
+    inputError: [
+      false,
+      {
+        setInputError: (_, { inputError }) => inputError,
+      },
+    ],
+    name: [
+      '',
+      {
+        setNameValue: (_, { name }) => name,
+      },
+    ],
+  },
+  selectors: ({ selectors }) => ({
+    canSubmit: [
+      () => [selectors.hasInputError, selectors.isLoading, selectors.name],
+      (hasInputError, isLoading, name) => !hasInputError && !isLoading && name.length > 0,
+    ],
+    hasInputError: [() => [selectors.inputError], (inputError) => inputError !== false],
```

Comment:
question: This returns true if inputError is an empty string, is that expected behavior?

Reply:
from my perspective I believe that should be the case. Now making it needing to be either strictly null or a string should enforce only two states of inputError.